codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -82,66 +96,75 @@ let make = (
           ~count=maxAddrInPartition,
         )
 
-      partitions :=
-        partitions.contents->List.add(
-          FetchState.makeRoot(~endBlock)(
-            ~staticContracts=[],
-            ~dynamicContractRegistrations=dynamicContractRegistrationsChunk,
-            ~startBlock,
-            ~logger,
-            ~isFetchingAtHead=false,
-          ),
-        )
+      let dynamicContractPartition = FetchState.makeRoot(~endBlock)(
+        ~staticContracts=[],
+        ~dynamicContractRegistrations=dynamicContractRegistrationsChunk,
+        ~startBlock,
+        ~logger,
+        ~isFetchingAtHead=false,
+      )
+      setNextPartition(dynamicContractPartition)
     }
-    partitions.contents
   }
 
   if Env.saveBenchmarkData {
     Benchmark.addSummaryData(
       ~group="Other",
       ~label="Num partitions",
-      ~value=partitions->List.size->Int.toFloat,
+      ~value=partitions->Js.Dict.keys->Array.length->Int.toFloat,
     )
   }
 
-  {maxAddrInPartition, partitions, endBlock, startBlock, logger}
+  {
+    maxAddrInPartition,
+    newestPartitionIndex: getNewestPartitionIndex(),
+    partitions,
+    endBlock,
+    startBlock,
+    logger,
+  }
 }
 
 let registerDynamicContracts = (
-  {partitions, maxAddrInPartition, endBlock, startBlock, logger}: t,
-  ~registeringEventBlockNumber,
-  ~registeringEventLogIndex,
-  ~dynamicContractRegistrations,
+  {partitions, newestPartitionIndex, maxAddrInPartition, endBlock, startBlock, logger}: t,
+  dynamicContractRegistration: FetchState.dynamicContractRegistration,
   ~isFetchingAtHead,
 ) => {
-  let partitions = switch partitions {
-  | list{head, ...tail} if head->FetchState.getNumContracts < maxAddrInPartition =>
+  let newestPartition = partitions->Js.Dict.unsafeGet(newestPartitionIndex->Int.toString)
```

Comment:
Not sure if I should rather do a runtime check here and explicitly fail if the partition does not exist.

Reply:
Let's do this, it won't harm.

Reply:
I think `unsafeGet` is good, when we 100% sure that the item will be present in the context of the function. But if we're talking about the whole system, it's better to do a check. Especially when it doesn't cost us anything.